stack/indexer/indexer-security-init.sh

```diff
@@ -12,40 +12,76 @@ HOST=""
 INSTALL_PATH="/usr/share/wazuh-indexer"
 CONFIG_PATH="/etc/wazuh-indexer"
 OPTIONS="-icl -nhnv"
+WAZUH_INDEXER_ROOT_CA="$(cat /etc/wazuh-indexer/opensearch.yml 2>&1 | grep http.pemtrustedcas | sed 's/.*: //')"
+WAZUH_INDEXER_ADMIN_PATH="$(dirname "${WAZUH_INDEXER_ROOT_CA}" 2>&1)"
+SECURITY_PATH="${INSTALL_PATH}/plugins/opensearch-security"
+CONFIG_FILE="${CONFIG_PATH}/opensearch.yml"
 
 # -----------------------------------------------------------------------------
 
 getNetworkHost() {
-    HOST=$(grep -hr "network.host:" "${CONFIG_PATH}"/opensearch.yml)
-    NH="network.host: "
-    HOST="${HOST//$NH}"
-
-    # Allow to find ip with an interface
-    if [[ "${HOST}" =~ _.*_ ]]; then
-        interface="${HOST//_}"
-        HOST=$(ip -o -4 addr list ${interface} | awk '{print $4}' | cut -d/ -f1)
-    fi
 
-    if [ "${HOST}" = "0.0.0.0" ]; then
-        HOST="127.0.0.1"
+    if [ -f "${CONFIG_FILE}" ]; then   
+        HOST=$(grep -hr "network.host:" "${CONFIG_FILE}" 2>&1)
+        NH="network.host: "
+        HOST="${HOST//$NH}"
+
+        # Allow to find ip with an interface
+        PATTERN="^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$"
+        if [[ ! "${HOST}" =~ ${PATTERN} ]]; then
+            interface="${HOST//_}"
+            HOST=$(ip -o -4 addr list "${interface}" | awk '{print $4}' | cut -d/ -f1)
+        fi
+
+        if [ "${HOST}" = "0.0.0.0" ]; then
+            HOST="127.0.0.1"
+        fi
     fi
+
 }
 
 # -----------------------------------------------------------------------------
+getPort() {
+
+    if [ -f "${CONFIG_FILE}" ]; then
+        PORT=$(grep -hr 'transport.tcp.port' "${CONFIG_FILE}" 2>&1)
+        if [ "${PORT}" ]; then
+            PORT=$(echo "${PORT}" | cut -d' ' -f2 | cut -d'-' -f1)
+        else
+            PORT="9300"
+        fi
+    fi
 
-PORT=$(grep -hr 'transport.tcp.port' "${CONFIG_PATH}/opensearch.yml")
-if [ "${PORT}" ]; then
-    PORT=$(echo ${PORT} | cut -d' ' -f2 | cut -d'-' -f1)
-else
-    PORT="9300"
-fi
-
+}
 # -----------------------------------------------------------------------------
 
 securityadmin() {
-SECURITY_PATH="${INSTALL_PATH}/plugins/opensearch-security"
 
-OPENSEARCH_PATH_CONF="${CONFIG_PATH}" JAVA_HOME="${INSTALL_PATH}/jdk" runuser wazuh-indexer --shell="/bin/bash" --command="${SECURITY_PATH}/tools/securityadmin.sh -cd ${SECURITY_PATH}/securityconfig -cacert ${CONFIG_PATH}/certs/root-ca.pem -cert ${CONFIG_PATH}/certs/admin.pem -key ${CONFIG_PATH}/certs/admin-key.pem -h ${HOST} -p ${PORT} ${OPTIONS}"
+    getPort
+
+    if [ ! -f "${CONFIG_FILE}" ]; then
+        echo "ERROR: it was not possible to find ${CONFIG_FILE}"
+        exit 1
```

Comment:
Move to the Line 10

Reply:
Done https://github.com/wazuh/wazuh-packages/pull/1441/commits/c88e0779f55f2646faab03b81d4c9a9e5a7fa551